lib/runtime-core/src/sys/windows/memory.rs

```diff
@@ -110,6 +110,13 @@ impl Memory {
         let size = round_up_to_page_size(range_end - range_start, page_size);
         assert!(size <= self.size);
 
+        // Allocating a memory of size 0 raises an error on Windows.
+        if size == 0 {
+            self.protection = Protect::None;
```

Comment:
Maybe we should just `self.protection = protect` here, and return `()`. I'm not sure what is best. The function receives a `protect` argument, and the user expects the memory to get this protection variant, not `None`.

Reply:
I think we should do `self.protection = protect`. This will shield better for the future.

```suggestion
            self.protection = protect;
```

Reply:
I did the change and tests started to fail?

Reply:
So maybe we shouldn't change for `protect` :-p.

Reply:
```suggestion
            self.protection = Protect::None;
```